src/Core/ConditionManager.php

```diff
@@ -13,16 +13,6 @@ class ConditionManager extends CoreConditionManager {
 
   /**
    * {@inheritdoc}
-   *
-   * @return \Drupal\rules\Core\RulesConditionInterface|\Drupal\Core\Condition\ConditionInterface
```

Comment:
This should not be removed. It's an helpful type-hint!

Reply:
OK, I was wondering about that. I guess the coder rule is intended so that you do not duplicate info that is also in the ancestor function which the inheritdoc is pointing to. So either we write out the full comment block without having {inheritdoc} or I can exclude that sniff from this specific file. I traced the inheritance from ConditionManager -> DefaultPluginManager -> PluginManagerBase where getDiscovery(), which just has 
```
/**
 * Gets the plugin discovery.
 *
 * @return \Drupal\Component\Plugin\Discovery\DiscoveryInterface
 */
```
Let me know your preference and I'll make the change.

Reply:
Ah, I was looking at the following function getDiscovery(). Sorry. The removal of the return is actually misleading, because the patch has split two functions. What is actually happening is the entire function is being deleted as a "useless override".